includes/Core/Modules/Module.php

```diff
@@ -817,15 +821,29 @@ protected function exception_to_error( Exception $e, $datapoint ) {
 			if ( isset( $errors[0]['reason'] ) ) {
 				$reason = $errors[0]['reason'];
 			}
+		} elseif ( $e instanceof Google_Proxy_Code_Exception ) {
+			$status        = 401;
```

Comment:
Technically, the message from the exception is actually the error code here (`WP_Error` code and message are typically both strings, where exceptions have numeric codes), i.e. in this clause specifically we should also set `$code = $message` and afterwards get the _actual_ message via `OAuth_Client::get_error_message`.

Reply:
Yep, updated.